client/post-editor/post-editor.jsx

```diff
@@ -228,9 +244,39 @@ export const PostEditor = React.createClass( {
 	},
 
 	toggleNestedSidebar: function( nestedSidebar ) {
+		if (
+			this.state.nestedSidebar === NESTED_SIDEBAR_REVISIONS &&
+			this.state.nestedSidebar !== nestedSidebar
+		) {
+			this.selectRevision( null );
```

Comment:
Seems like we're mixing concerns here, are we not? Any "extension" of the editor can make use of nested sidebars, so this logic is expected to be generic, yet there is now logic specific to revisions; in addition, it can be argued that this scatters the state-keeping logic of revisions in more places, thus hurting maintainability.

Can we find a more decoupled (or at least indirect) way to achieve the same result?